fingerprint/src/main/java/com/fingerprintjs/android/fingerprint/info_providers/SensorsInfoProvider.kt

```diff
@@ -18,18 +19,16 @@ public interface SensorDataSource {
 }
 
 internal class SensorDataSourceImpl(
-    private val sensorManager: SensorManager
+    private val sensorManager: SafeLazy<SensorManager>
 ) : SensorDataSource {
     override fun sensors(): List<SensorData> {
-        return executeSafe(
-            {
-                sensorManager.getSensorList(Sensor.TYPE_ALL).map {
+        return safe {
+                sensorManager.getOrThrow().getSensorList(Sensor.TYPE_ALL)!!.filterNotNull().map {
                     SensorData(
-                        it.name,
-                        it.vendor
+                        it.name!!,
```

Comment:
If one sensor will have null-name or vendor – then all the list becomes empty?

Reply:
Correct. This way the previous behaviour is preserved.